Sources/SwiftCrossUI/Environment/Environment.swift

```diff
@@ -36,21 +36,26 @@
 /// ```
 @propertyWrapper
 public struct Environment<Value>: DynamicProperty {
-    var keyPath: KeyPath<EnvironmentValues, Value>
+    var keyPath: KeyPath<EnvironmentValues, Value>?
+    var environmentKey: (any EnvironmentKey.Type)?
```

Comment:
Introduce an enum to encode the requirement that we want one property or the other to be non-nil (not neither, and not both). This enum would just be an internal detail.

```swift
enum EnvironmentLocation<Value> {
    case keyPath(KeyPath<EnvironmentValues, Value>)
    case environmentKey(any EnvironmentKey.Type)
}
```

You can then replace these two properties with a single `var location: EnvironmentLocation<Value>` property, and then replace the existing if lets with switch statements.